modules/listeners/models/transfer.js

```diff
@@ -25,13 +25,27 @@ module.exports = (osseus) => {
         // TODO here probably need to notify someone/somehow
       } else if (!knownFrom) {
         osseus.logger.info(`Transfer event from unknown address (deposit) - from: ${from}, to: ${to}, amount: ${amount.toNumber()}`)
-
         const bctx = await osseus.lib.BlockchainTransaction.deposit(data.transactionHash, from, to, token, amount)
         osseus.logger.info(`Created blockchain deposit transaction: ${JSON.stringify(bctx)}`)
         const tx = await osseus.lib.Transaction.deposit({accountAddress: to, currency: token}, amount, bctx.id)
         osseus.logger.info(`Created offchain deposit transaction: ${JSON.stringify(tx)}`)
-
         // TODO here probably need to notify someone/somehow
+      } else {
+        osseus.logger.info(`Transfer event with known addresses - from: ${from}, to: ${to}, amount: ${amount.toNumber()}`)
+        data.type = 'TRANSFER'
+        data.state = 'CONFIRMED'
+        data.meta = {from: from, to: to, token: token, amount: amount.toString()}
+        const update = {
+          $set: data,
+          $setOnInsert: {
+            known: false
+          }
+        }
+        const tx = await osseus.db_models.bctx.update({hash: data.transactionHash}, update)
+        osseus.logger.debug(`Updated blockchain transaction: ${JSON.stringify(tx)}`)
+
+        await osseus.utils.updateBlockchainBalance(from, token)
+        await osseus.utils.updateBlockchainBalance(to, token)
```

Comment:
@bejavu do we need to update the off-chain balance as well? we will have a difference at this point...

Reply:
We need to keep track on the `known: false` transactions and notify \ halt the system if it happen

Reply:
Added a ticket for this